src/fides/api/ops/service/connectors/saas/connector_registry_service.py

```diff
@@ -235,3 +289,74 @@ def update_saas_instance(
     connection_config.update_saas_config(db, SaaSConfig(**config_from_template))
 
     upsert_dataset_config_from_template(db, connection_config, template, template_vals)
+
+
+def register_custom_functions(script: str) -> None:
```

Comment:
Uses a combination of RestrictedPython and ZopeGuards (both from the Zope Foundation) to load custom code in a restricted environment. The restrictions work by limiting which Python functions (builtins) the custom script has access to. The out-of-the-box settings are too restrictive so we had to make a few exceptions:

- Allowing the use of type annotations (via CustomRestrictingNodeTransformer)
- Allowing all imports by default and excluding some unsafe modules (custom_guarded_import)
- Allowing `getitem`
- Allowing `staticmethod`